### ISSUE TITLE
pipenv graph does not show all top-level packages

### ISSUE THREAD

**Issue by thernstig** [pypa/pipenv#3460]
### Issue description

My Pipfile contains the following:

```toml
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]
requests = "*"

[requires]
python_version = "2.7"
```

Pipenv graph then shows the output:

```sh
requests==2.21.0
  - certifi [required: >=2017.4.17, installed: 2018.11.29]
  - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
  - idna [required: >=2.5,<2.9, installed: 2.8]
  - urllib3 [required: >=1.21.1,<1.25, installed: 1.24.1]
```

I add another package under [dev-packages]:

```toml
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]
sphinx-rtd-theme = "*"

[packages]
requests = "*"

[requires]
python_version = "2.7"
```

Since `sphinx-rtd-theme` has a dependency towards `requests`, the graph now only shows `requests` underneath `sphinx-rtd-theme`.

```sh
sphinx-rtd-theme==0.4.2
  - sphinx [required: Any, installed: 1.8.3]
    - alabaster [required: >=0.7,<0.8, installed: 0.7.12]
    - babel [required: >=1.3,!=2.0, installed: 2.6.0]
      - pytz [required: >=0a, installed: 2018.9]
    - docutils [required: >=0.11, installed: 0.14]
    - imagesize [required: Any, installed: 1.1.0]
    - Jinja2 [required: >=2.3, installed: 2.10]
      - MarkupSafe [required: >=0.23, installed: 1.1.0]
    - packaging [required: Any, installed: 18.0]
      - pyparsing [required: >=2.0.2, installed: 2.3.1]
      - six [required: Any, installed: 1.12.0]
    - Pygments [required: >=2.0, installed: 2.3.1]
    - requests [required: >=2.0.0, installed: 2.21.0]
      - certifi [required: >=2017.4.17, installed: 2018.11.29]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.9, installed: 2.8]
      - urllib3 [required: >=1.21.1,<1.25, installed: 1.24.1]
    - setuptools [required: Any, installed: 40.6.3]
    - six [required: >=1.5, installed: 1.12.0]
    - snowballstemmer [required: >=1.1, installed: 1.2.1]
    - sphinxcontrib-websupport [required: Any, installed: 1.1.0]
    - typing [required: Any, installed: 3.6.6]
```

### Expected result

I expected `requests` to also be shown at the top level, as it is a top-level dependency. The reason why I expect this is that other dependencies are shown multiple times. Look at for example `six` shown twice above.

I assume that it is reasonable to show main packages at the top and then also as again if it is a dependency of other packages.

### Steps to replicate

1. Start in a fresh directory and call `pipenv install requests`
2. Check `pipenv graph`
3. Run `pipenv install --dev sphinx-rtd-theme`
4. Check `pipenv graph`

**Comment by frostming** [comment]
`pipenv graph` shows the dependency graph of the current working set, which has nothing to do with what is defined in Pipfile. The graph is presented so that any package name only appears once.

I think we should just keep what it is. Any thoughts? @jxltom @techalchemy

**Comment by jxltom** [comment]
@frostming The ```pipenv graph``` has nothing to do with ```Pipfile``` and ```Pipfile.lock```. It will even show the packages installed by pip directly. Though it is not hard to implement, it might be confusing if we show packages according to the content of Pipfile since users may wonder why some packages which are not defined in Pipfile are also in the top level. One more thing is that ```pipenv graph``` won't ensure the project which means the Pipfile may be broken or empty. If we'd like to implement it, there are probably a few corner cases to deal with. So for now I prefer keeping it as it is.

**Comment by jxltom** [comment]
Just one more thing to say, @thernstig please feel free to open a PEEP https://github.com/pypa/pipenv/blob/master/peeps/PEEP-000.md if your still strongly recommend this feature, and we can discuss further under that PEEP 😃

**Comment by thernstig** [comment]
Appreciate the answers. I have a few questions on how to work around this then.

> The graph is presented in the way so that any package name appears only once.

If you look at the package `six` above it is shown twice. This is true for any other package which is a "sub-dependency". Since `requests` is a main dependency for my project, I would have expected the graph show it both at the top as well as burrowed in the dependency tree under `sphinx-rtd-theme`.

The reason why I want to use this is due to that we have a complicated process to approve the licenses used by packages that are used in production. So I want to get a list of all python packages that only used in production. `pipenv graph` has no such support. A work-around is to only issue a `pipenv install` and then show the graph, but this is not possible in our case on our CI servers due to "reasons". I can work around all this with some complex parsing. But my main point is still if `pipenv graph` should not show these packages.

It would make more sense to me at least to show the package at all levels, at least as a novice without understanding of the inner workings of the dependency resolution 😄

**Comment by jxltom** [comment]
So actually you would like to let the pipenv support ```pipenv graph --default```, ```pipenv graph --dev``` or something like that.

Moving declared packages in Pipfile to the top level is just a workaround which makes your own parsing easier.

Is my understanding right?

**Comment by chivalry** [comment]
I just throw my two cents in and mention that I almost created another issue for this but GitHub nicely found this thread for me.

Given what I see in `pipenv graph` I would assume that any top-level packages are going to show what they depend upon and packages installed because of a dependency would not. In my case, `Flask-SQLAlchemy` is a top-level package but it only appears as a dependency in `pipenv graph`, under `Flask-Migrate` (also a top-level dependency.

And I'll admit, I only started using `pipenv` today, so there every chance there's an underlying philosophy I'm unfamiliar with. I'm still reading the docs. :)

**Comment by Pastromhaug** [comment]
Hiya, just wanna add that I was also very confuse about this. I totally assumed that any top-level packages described in my `pipfile` would be the top-level packages in the graph and had no idea what I was looking at.
